code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -13,481 +13,374 @@ What are the archived variables for?
+	//Returns: a string indicating what check failed, or "" if check passes
 
-	///Creates new, identical gas mixture
-	///Returns: duplicate gas mixture
-/datum/gas_mixture/proc/copy()
-	var/list/cached_gases = gases
-	var/datum/gas_mixture/copy = new type
-	var/list/copy_gases = copy.gases
+/datum/gas_mixture/proc/react(datum/holder)
+	//Performs various reactions such as combustion or fusion (LOL)
+	//Returns: 1 if any reaction took place; 0 otherwise
 
-	copy.temperature = temperature
-	for(var/id in cached_gases)
-		ADD_GAS(id, copy.gases)
-		copy_gases[id][MOLES] = cached_gases[id][MOLES]
+/datum/gas_mixture/proc/__remove()
+/datum/gas_mixture/remove(amount)
+	var/datum/gas_mixture/removed = new type
+	__remove(removed, amount)
 
-	return copy
+	return removed
 
-	///Copies variables from sample, moles multiplicated by partial
-	///Returns: 1 if we are mutable, 0 otherwise
-/datum/gas_mixture/proc/copy_from(datum/gas_mixture/sample, partial = 1)
-	var/list/cached_gases = gases //accessing datum vars is slower than proc vars
-	var/list/sample_gases = sample.gases
+/datum/gas_mixture/proc/__remove_ratio()
+/datum/gas_mixture/remove_ratio(ratio)
+	var/datum/gas_mixture/removed = new type
+	__remove_ratio(removed, ratio)
 
-	//remove all gases not in the sample
-	cached_gases &= sample_gases
+	return removed
 
-	temperature = sample.temperature
-	for(var/id in sample_gases)
-		ASSERT_GAS(id,src)
-		cached_gases[id][MOLES] = sample_gases[id][MOLES] * partial
+/datum/gas_mixture/copy()
+	var/datum/gas_mixture/copy = new type
+	copy.copy_from(src)
 
-	return 1
+	return copy
 
-	///Copies all gas info from the turf into the gas list along with temperature
-	///Returns: 1 if we are mutable, 0 otherwise
-/datum/gas_mixture/proc/copy_from_turf(turf/model)
+/datum/gas_mixture/copy_from_turf(turf/model)
 	parse_gas_string(model.initial_gas_mix)
 
 	//acounts for changes in temperature
 	var/turf/model_parent = model.parent_type
 	if(model.temperature != initial(model.temperature) || model.temperature != initial(model_parent.temperature))
-		temperature = model.temperature
+		set_temperature(model.temperature)
 
 	return 1
 
-	///Copies variables from a particularly formatted string.
-	///Returns: 1 if we are mutable, 0 otherwise
-/datum/gas_mixture/proc/parse_gas_string(gas_string)
-	gas_string = SSair.preprocess_gas_string(gas_string)
-
-	var/list/gases = src.gases
+/datum/gas_mixture/parse_gas_string(gas_string)
 	var/list/gas = params2list(gas_string)
 	if(gas["TEMP"])
-		temperature = text2num(gas["TEMP"])
+		set_temperature(text2num(gas["TEMP"]))
 		gas -= "TEMP"
-	gases.Cut()
+	clear()
 	for(var/id in gas)
 		var/path = id
 		if(!ispath(path))
 			path = gas_id2path(path) //a lot of these strings can't have embedded expressions (especially for mappers), so support for IDs needs to stick around
-		ADD_GAS(path, gases)
-		gases[path][MOLES] = text2num(gas[id])
+		set_moles(path, text2num(gas[id]))
+	archive()
```

Comment:
Seems like a good time to ask how archive() works in monstermos, as I've seen garbage_collect(), which calls it, removed in a few places. I assume that's handled internally now?

Reply:
gc is handled internally now yes